### PR TITLE
chore: added analytics for badge interactions

### PR DESCRIPTION
**Pull request by fcolarich**
## What does this PR change?

Added new analytics to track profile and badge interactions
...

## How to test the changes?

Do badge interactions and then check if they are being recorded correctly I guess? :hmm

### CODE REVIEW

File: Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/EventBased/PassportAnalytics.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                {
+                    { "receiver_id", userId },
+                });
+                return;
+            }
+
+            analytics.Track(AnalyticsEvents.Profile.PASSPORT_OPENED, new JsonObject
+            {
+                { "receiver_id", userId }
+            });
```
Comment (popuz) [inline Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/EventBased/PassportAnalytics.cs:59]: you can simplify it by presenting 
`string event = isOwnPassposrt? (AnalyticsEvents.Profile.OWN_PROFILE_OPENED, AnalyticsEvents.Profile.PASSPORT_OPENED);`
and then using it in just one event

File: Explorer/Assets/DCL/Passport/Modules/Badges/BadgesDetails_PassportModuleController.cs [hunk 2]
```diff
@@ -50,6 +52,7 @@ public BadgesDetails_PassportModuleController(
             badgeInfoController = new BadgeInfo_PassportModuleSubController(badgeInfoModuleView, webRequestController, badgesAPIClient, passportErrorsController);
             badgeDetailsCardsController = new BadgeDetailsCards_PassportModuleSubController(view, webRequestController, badgesCategoriesController, badgeInfoController);
 
+            badgeDetailsCardsController.OnBadgeSelected += (id) => OnBadgeSelected?.Invoke(id);
```
Comment (popuz) [inline Explorer/Assets/DCL/Passport/Modules/Badges/BadgesDetails_PassportModuleController.cs:55]: every time this event is triggered, a new delegate will be created. It is small allocations, but let's avoid it

**Review by popuz — CHANGES_REQUESTED** [review]
code looks good, but events are not working. Please, check it by enabling ANALYTICS log section in the ReportHandling config and verifying behavior in the Unity console

File: Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/EventBased/PassportAnalytics.cs (hunk 1 above)
Comment (fcolarich) [reply Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/EventBased/PassportAnalytics.cs:59]: goood idea!

File: Explorer/Assets/DCL/Passport/Modules/Badges/BadgesDetails_PassportModuleController.cs (hunk 2 above)
Comment (fcolarich) [reply Explorer/Assets/DCL/Passport/Modules/Badges/BadgesDetails_PassportModuleController.cs]: fixed!

**Review by popuz — APPROVED** [review]
checked together - analytics are coming to the Segment 🟢

**Review by aixaCode — APPROVED** [review]

**Review by DafGreco — APPROVED** [review]
Validation done on Windows and Macos ✅ 

- Notifications
- Badges 
- Passport

**Review by anicalbano — APPROVED** [review]

**Pull request merged**